### ISSUE TITLE
add resin config generate --ethernet and --wifi options for noninteractive cli

### ISSUE THREAD

**Issue by hh** [resin-io/resin-cli#410]
Currently ```resin config generate``` does not allow specifying all necessary options via the cli.
It requires navigating with arrow keys interactively via console rather than specifying parameters.

```
$ resin config generate --app iichip --output config.json 
? Network Connection (Use arrow keys)
❯ ethernet 
  wifi 
```

This poses a problem to automation pipelines which currently need a work around using something like expect:

https://gitlab.ii.org.nz/iichip/resin-cli/blob/latest/config.exp
```
#!/usr/bin/expect -f
spawn resin config generate --app "$env(APP)" --output config.json
match_max 100000
expect "ethernet"
send -- "\r"
expect "Check for updates every X minutes"
send -- "\r"
expect eof
```
Our work around just hit's enter twice to enable ethernet, but adding wifi configuration to the pipeline would be great.

@ii wrapped this expect into a [Docker container](https://gitlab.ii.org.nz/iichip/resin-cli/blob/latest/Dockerfile) to allow us to [generate the config.json during our resin-os image bulid](https://gitlab.ii.org.nz/iichip/resin-chip/blob/fastbuild/.gitlab-ci.yml#L46
):

```
docker pull iichip/resin-cli
docker run -e EMAIL="$EMAIL" \
           -e APP="$APP" \
           -e PASSWORD="$PASSWORD" iichip/resin-cli > config.json
```

<img src="https://frontapp.com/assets/img/icons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_1ihj)

**Comment by jviotti** [comment]
Hey there,

Thanks for the suggestion, it sounds like a good addition. The problem is that the set of question we ask to be able to generate a `config.json` comes from an external service (the CLI blindly asks such questions), so hardcoding `--wifi`/`--ethernet` options would not be a possible way forward. As a consequence, we have to add some way for allowed options to be computed at runtime, or put a generic option what you can pass any set of key/value combinations that will be applied on top of the retrieved question list. We could also accept a path to a json file containing certain defaults, like `networkType: 'wifi'`. Would that suit your needs?

**Comment by hh** [comment]
Paired with a --show-opts and --generate-opt-file fill-me-out.json/yaml I think we have a winning combination!

Is there an API test suite or documentation area that addresses config generate and friends?
A quick look at https://docs.resin.io/runtime/data-api/ didn't reveal anything.

**Comment by jviotti** [comment]
Sounds good! I'll try to allocate some time for this soon. In order to get unblocked, you could use https://github.com/resin-io-modules/resin-device-config, which is the low-level module that generates configuration to create a simple script that does what you want. You could also take a look at https://github.com/resin-io/resin-config-json, which is the module that injects `config.json` to the device, in case you need that as well!

**Comment by imrehg** [comment]
In general I get quite a few requests for completely non-interactive `resin config generate` (so it's possible to use the script automated). Would that be possible to include in the near-term updates? Would help with things like the workflow we've shown in our [preloading and preprovisioning blogpost](https://resin.io/blog/advanced-device-provisioning-workflow-for-large-fleets-preloading-and-pre-provisioning/)

**Comment by pimterry** [comment]
Fixed and released in 6.9.0, see the docs for `config generate` for the full details.

Example:

```
resin config generate --app MyApp --network wifi --wifiSsid mySsid --wifiKey abcdefgh --appUpdatePollInterval 1
```
